src/officehours_api/serializers.py

```diff
@@ -260,10 +260,14 @@ def validate_queue(self, queue):
 
     def validate(self, attrs):
         '''
-        Ensure the assignee is a host.
+        Ensure the assignee is a host,
+        and the meeting's backend type is in the queue's allowed backend type.
         '''
         queue = self.instance.queue if self.instance else attrs["queue"]
         hosts = queue.hosts.all()
         if attrs.get("assignee") and attrs["assignee"] not in hosts:
             raise serializers.ValidationError("Assignee must be a host!")
+        backend_type = attrs["backend_type"]
+        if backend_type not in queue.allowed_backends:
+            raise serializers.ValidationError(f"{backend_type} is not one of the queue's allowed backend types ({queue.allowed_backends})")
```

Comment:
So I did a little testing, and I found some issues with this. I think we need to use the same syntax we did for `assignee`, where we check whether it exists using `attrs.get(...)` before we use it. There are a couple ways in the application that a meeting can be `PATCH`ed or updated, and it's not a given that the `backend_type` will be included. Once this is modified, it probably makes sense to add a couple test cases for the patching as well.

Let me know if you need assistance formulating those test cases. There is a PATCH endpoint for meetings in the Swagger UI, and you can get the meeting IDs from the Django admin site or the database.

Reply:
We have modified the code and added a couple of tests in the test file. We are considering adding a test case (`test_patch_backend_type_invalid()`) in which the meeting is valid first (in person), then we patch it to an invalid type (zoom), and we would get the expected error message. However, since we don't have the credential of zoom type, the error message wouldn't be the one we expected, so the test would not pass until the credential is included. Should we leave the test here or just delete it?

Reply:
Looks like you made the right change to the validation. Let me take a look at the tests and I'll let you know if I think they are reasonable. Yes, it's probably a bit weird to test without both backend types enabled.

Reply:
I figured out one naive way to solve this is to run the test and set the environment variable at runtime. So it could pass all tests if we just run `docker-compose run -e ZOOM_CLIENT_ID="test" -e ZOOM_CLIENT_SECRET="test" web python manage.py test officehours_api.tests.MeetingSerializerTestCase` by setting the ZOOM environment variables on the go.